fusestore/store/src/executor/meta_handlers.rs

```diff
@@ -23,40 +23,61 @@ use common_flights::meta_api_impl::GetDatabaseAction;
 use common_flights::meta_api_impl::GetDatabaseActionResult;
 use common_flights::meta_api_impl::GetTableAction;
 use common_flights::meta_api_impl::GetTableActionResult;
+use common_metatypes::Database;
+use common_metatypes::Table;
 use log::info;
 
 use crate::executor::action_handler::RequestHandler;
 use crate::executor::ActionHandler;
-use crate::protobuf::CmdCreateDatabase;
-use crate::protobuf::CmdCreateTable;
-use crate::protobuf::Db;
-use crate::protobuf::Table;
+use crate::meta_service::cmd::Cmd::CreateDatabase;
+use crate::meta_service::cmd::Cmd::CreateTable;
+use crate::meta_service::cmd::Cmd::DropDatabase;
+use crate::meta_service::cmd::Cmd::DropTable;
+use crate::meta_service::AppliedState;
+use crate::meta_service::LogEntry;
 
 // Db
-
 #[async_trait::async_trait]
 impl RequestHandler<CreateDatabaseAction> for ActionHandler {
     async fn handle(
         &self,
         act: CreateDatabaseAction,
     ) -> common_exception::Result<CreateDatabaseActionResult> {
         let plan = act.plan;
-        let mut meta = self.meta.lock();
-
-        let cmd = CmdCreateDatabase {
-            db_name: plan.db,
-            db: Some(Db {
-                // meta fills it
-                db_id: -1,
-                ver: -1,
-                table_name_to_id: HashMap::new(),
-                tables: HashMap::new(),
-            }),
+        let db_name = &plan.db;
+
+        let cr = LogEntry {
+            txid: None,
+            cmd: CreateDatabase {
+                name: db_name.clone(),
+                if_not_exists: plan.if_not_exists,
+                db: Database {
+                    database_id: 0,
+                    tables: HashMap::new(),
+                },
+            },
         };
 
-        let database_id = meta.create_database(cmd, plan.if_not_exists)?;
+        let rst = self.meta_node.write(cr).await.map_err(|e| {
+            ErrorCode::MetaNodeInternalError(format!("not a Database result: {}", e.to_string()))
```

Comment:
What does it mean `not a Database result`?

Reply:
Here's a quote from a previous writeup, `not a Database result` means it's not a database operation error, maybe this error occurred by meta service or something else.

Reply:
I think the `not a Database result` is a bit misleading.
Here it may be a network error or else.

Reply:
I think the `not a Database result` is a bit misleading.
Here it may be a network error or else.